src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -226,6 +226,13 @@ protected void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArguments<At
                 return;
             }
 
+            if (arguments.Attribute.IsTargetAttribute(this, AttributeDescription.ExperimentalAttribute)
+                && !SyntaxFacts.IsValidIdentifier((string?)arguments.Attribute.CommonConstructorArguments[0].ValueInternal))
+            {
+                arguments.Diagnostics.DiagnosticBag.Add(ErrorCode.ERR_InvalidExperimentalDiagID, arguments.AttributeSyntaxOpt.Location);
```

Comment:
>arguments.AttributeSyntaxOpt.Location [](http://example.com/codeflow?start=97&length=37)

I think we have a helper that finds syntax for specific argument when there is one